app/create_file.py

```diff
@@ -1 +1,56 @@
-# write your code here
+from datetime import datetime
+from sys import argv
+from os import mkdir, chdir
+
+filename = ""
+directories = []
+
+if "-f" in argv:
+    index = 0
+
+    for _ in range(len(argv) - 1):
+        index += 1
+        if argv[_: _ + 1] == ["-f"]:
+            break
+
+    filename = argv[index]
+
+    if "-d" in argv:
+        index2 = 0
+
+        for _ in range(len(argv) - 1):
+            index2 += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index2:index - 1]
+else:
+    filename = "file.txt"
+
+    if "-d" in argv:
+        index = 0
+
+        for _ in range(len(argv) - 1):
+            index += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index:len(argv)]
```

Comment:
This code block has a room for improvement:
1. Multiple nested if else statements and code loops, they make this code hard to understand.
2. Use of `_` as variable during iteration and incrementing index (meaningfully use loop output instead of skipping value to `_`).
3. `index` and `index2` variable names are not descriptive. 
4. Code duplication in case `-d` argument is present in `argv`.
5. Defining the filename in the middle of the code processing, this operation should be performed outside of the argument parsing.

That is why I would suggest to separate the logic of arguments parsing with the use of `argparse`, like the following:

```suggestion
parser = argparse.ArgumentParser()
parser.add_argument("-d", dest="dir", nargs="*")
parser.add_argument("-f", dest="filename")

parsed_args = parser.parse_args()
filename = parsed_args.filename
directories = parsed_args.dir
```